sdk/data/azcosmos/cosmos_client_retry_policy.go

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcosmos
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+)
+
+type clientRetryPolicy struct {
+	gem                    *globalEndpointManager
+	retryCount             int
+	sessionRetryCount      int
+	preferredLocationIndex int
+}
+
+const maxRetryCount = 10 // this number needs to be higher, keeping low for testing purposes right now
+const defaultBackoff = 1
+
+func (p *clientRetryPolicy) Do(req *policy.Request) (*http.Response, error) {
+	response, err := req.Next() // err can happen in weird scenarios (connectivity, etc) - need to test
```

Comment:
Remember to include the `err` for the DNS error that Joel shared? https://pkg.go.dev/net#DNSError

Reply:
yes I just haven't gotten to this, finishing the main test scenarios first